src/main/src/main/java/org/geoserver/logging/Log4jUtils.java

```diff
@@ -0,0 +1,705 @@
+/* (c) 2014 - 2016 Open Source Geospatial Foundation - all rights reserved
+ * (c) 2001 - 2013 OpenPlans
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.logging;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import org.apache.commons.io.IOUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.core.Appender;
+import org.apache.logging.log4j.core.LoggerContext;
+import org.apache.logging.log4j.core.appender.ConsoleAppender;
+import org.apache.logging.log4j.core.appender.FileAppender;
+import org.apache.logging.log4j.core.appender.RollingFileAppender;
+import org.apache.logging.log4j.core.config.Configuration;
+import org.apache.logging.log4j.core.config.ConfigurationSource;
+import org.apache.logging.log4j.core.config.DefaultConfiguration;
+import org.geoserver.logging.LoggingUtils.GeoToolsLoggingRedirection;
+import org.geoserver.platform.GeoServerResourceLoader;
+import org.geoserver.platform.resource.Paths;
+import org.geoserver.platform.resource.Resource;
+import org.geoserver.platform.resource.Resource.Type;
+import org.geoserver.platform.resource.Resources;
+import org.vfny.geoserver.global.ConfigurationException;
+
+/**
+ * Used to manage GeoServer logging facilities.
+ *
+ * <p>GeoTools logging is configured following {@link GeoToolsLoggingRedirection} policy, mapping
+ * {@link #GT2_LOGGING_REDIRECTION} policy to appropriate LoggingFactory responsible for routing
+ * java util logging api.
+ *
+ * <p>Use {@code -DGT2_LOGGING_REDIRECTION=Log4j} {@code -DRELINQUISH_LOG4J_CONTROL=true} {@code
+ * -Dlog4j.configuration=log4j.properties} to redirect GeoServer to use LOG4J API with
+ * log4j.properties configuration.
+ *
+ * <p>Prior to GeoSerer 2.21 the LOG4J library was included as the application default, to maintain
+ * an older data directory use {@code -DGT2_LOGGING_REDIRECTION=Log4j2
+ * -DRELINQUISH_LOG4J_CONTROL=true -Dlog4j.configuration=CUSTOM_LOGGING.properties}. This forces
+ * GeoServer to read your {@code CUSTOM_LOGGING.properties} Log4J 1.2 configuration (via Log4J
+ * bridge).
+ *
+ * <p>To troubleshoot use {@code org.apache.logging.log4j.simplelog.StatusLogger.level=DEBUG}, or
+ * adjust {@code log4j2.xml} file &lt;Configuration status=&quote;trace&quote;&gt;.
+ *
+ * @see org.geoserver.config.LoggingInfo
+ */
+public class Log4jUtils {
```

Comment:
A couple quick suggestions:

* Rename to LoggingUtilsDelegate (since this is kind of a delegate if I read the code correctly).
* No need to be a public class, package visibility is enough 
* Double check LoggingStartupContextListener as it should be called first (it is responsible for redirecting if needed).

Reply:
I renamed the class and changed it to package visibility.  There were no changes made to LoggingStartupContextListener  or that need to be made to that class.